modules/tensor_mechanics/src/materials/CZM/CZM3DCLaw.C

```diff
@@ -0,0 +1,169 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CZM3DCLaw.h"
+
+registerMooseObject("TensorMechanicsApp", CZM3DCLaw);
+
+template <>
+InputParameters
+validParams<CZM3DCLaw>()
+{
+  InputParameters params = validParams<CZMMaterialBase>();
+  params.addClassDescription("3DC cohseive law model, no damage");
```

Comment:
3D Coupled (3DC) cohesive law of Salehani and Irani with no damage

Reply:
ok